vimeo-networking/src/main/java/com/vimeo/networking/model/Interaction.java

```diff
@@ -22,16 +22,59 @@
 
 package com.vimeo.networking.model;
 
+import com.google.gson.annotations.SerializedName;
+
+import org.jetbrains.annotations.Nullable;
+
 import java.io.Serializable;
 import java.util.Date;
 
 /**
+ * This model object represents an Interaction.
  * Created by zetterstromk on 6/5/15.
  */
 public class Interaction implements Serializable {
 
     private static final long serialVersionUID = 2033767841952340400L;
+    private static final String STREAM_PURCHASED = "purchased";
+    private static final String STREAM_RESTRICTED = "restricted";
+    private static final String STREAM_AVAILABLE = "available";
+    private static final String STREAM_UNAVAILABLE = "unavailable";
+
+    public enum Stream {
+        @SerializedName(STREAM_PURCHASED)
+        PURCHASED(STREAM_PURCHASED),
+        @SerializedName(STREAM_RESTRICTED)
+        RESTRICTED(STREAM_RESTRICTED),
+        @SerializedName(STREAM_AVAILABLE)
+        AVAILABLE(STREAM_AVAILABLE),
+        @SerializedName(STREAM_UNAVAILABLE)
+        UNAVAILABLE(STREAM_UNAVAILABLE);
```

Comment:
Yeah, in general it would be great to do that, but the networking library is pretty absent of any kind of Java docs. Maybe we should take care of that in a general sweep when we update all of the classes with private getters and setters? What do you think?